### ISSUE TITLE
shadowtexXMipmap and shadowcolorXMipmap cannot work

### ISSUE THREAD

**Issue by GeForceLegend** [IrisShaders/Iris#2383]
### What happened?

After using below code, mipmap for those shadowtex and shadowcolor is still unable to use, making some shadow related stuff using mipmap become much more expensive. IMS should have a copy of my shader so he can test these thing.
```glsl
const bool shadowtex0Mipmap = true;
const bool shadowcolor0Mipmap = true;
const bool shadowcolor1Mipmap = true;
```

### Screenshots

Shadowcolor0 and shadowcolor0 only have 1 mip (or their mipmap is not generated)
![shadowcolor0](https://github.com/IrisShaders/Iris/assets/41003546/edc50e65-ec19-4e91-bcbc-534007a4be27)
![shadowcolor1](https://github.com/IrisShaders/Iris/assets/41003546/18515df3-6657-42e4-b49b-3a1d67eb30c7)

Shadowtex0 has mipmap generated, but looks like it's mipmap cannot get used
![shadowtex](https://github.com/IrisShaders/Iris/assets/41003546/2eabc2f6-588f-49d5-8178-43c1e1b96884)

### Log output

[latest.log](https://github.com/user-attachments/files/15963466/latest.log)

### Minecraft Version

1.20.6

### Iris Version

iris-1.7.1+mc1.20.6.jar

### Operating System

Windows10 Home 22H2 19045.4529

### What is your GPU?

NVIDIA GeForce RTX 4070

**Comment by GeForceLegend** [comment]
`shadowcolorXMipmap` now working correctly, but for some reasons, `shadowtexXMipmap` on some shaderpack still cannot work. Here is an example pack, I guess it has something to do with `shadowHardwareFiltering`

[shadowtex0_mipmap_test.zip](https://github.com/user-attachments/files/17789580/shadowtex0_mipmap_test.zip)

**Comment by GeForceLegend** [comment]
Looks get completely fixed in 1.8.1
